packages/system/src/Location/ListView/api.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  useQueryParams,
+  useQuery,
+  useOmSupplyApi,
+  UseQueryResult,
+  LocationsQuery,
+  QueryParamsState,
+  SortBy,
+  LocationSortInput,
+  LocationSortFieldInput,
+} from '@openmsupply-client/common';
+import { Location } from '../types';
+
+const toSortInput = (sortBy: SortBy<Location>): LocationSortInput => {
+  return { desc: sortBy.isDesc, key: sortBy.key as LocationSortFieldInput };
+};
+
+const locationsGuard = (locationsQuery: LocationsQuery) => {
+  if (locationsQuery.locations.__typename === 'LocationConnector') {
+    return locationsQuery.locations;
+  }
+
+  throw new Error(locationsQuery.locations.error.description);
+};
+
+export const useLocationList = (): UseQueryResult<
```

Comment:
I'm being lazy and haven't added pagination or filtering.. how many locations do people have? 🤷

Reply:
oh! I was clicking on the pagination control and not noticing that the list stayed the same! 🙄 
Agree.. there shouldn't be that many locations.